File: frontend/src/api/letter-verification.ts
import request from './request'
import type { ApiResponse } from './request'

/**
 * 函件验证结果接口
 */
export interface LetterVerificationResult {
  valid: boolean
  verifyStatus: 'VALID' | 'INVALID' | 'EXPIRED' | 'REVOKED' | 'NOT_FOUND'
  statusMessage: string
  applicationNo?: string
  letterTypeName?: string
  firmName?: string
  lawyerNames?: string
  targetUnit?: string
  matterName?: string
  approvedAt?: string
  printedAt?: string
  validUntil?: string
  verifyCount?: number
  remark?: string
}

/**
 * 函件验证记录接口（管理后台）
 */
export interface LetterVerificationRecord {
  id: number
  letterId: number
  applicationNo: string
  verificationCode: string
  letterType?: string
  letterTypeName?: string
  targetUnit?: string
  lawyerNames?: string
  firmName?: string
  matterName?: string
  approvedAt?: string
  printedAt?: string
  validUntil?: string
  remark?: string
  verifyCount: number
  lastVerifyAt?: string
  lastVerifyIp?: string
  status: string
  createdAt: string
  updatedAt: string
}

/**
 * 分页查询结果接口
 */
export interface VerificationListResponse {
  records: LetterVerificationRecord[]
  total: number
  page: number
  pageSize: number
  pages: number
}

/**
 * 统计数据接口
 */
export interface VerificationStatistics {
  total: number
  active: number
  expired: number
  revoked: number
}

// ========== 公开接口 ==========

/**
 * 验证函件真伪
 * @param applicationNo 函件申请编号
 * @param code 验证码
 */
export function verifyLetter(applicationNo: string, code: string): Promise<ApiResponse<LetterVerificationResult>> {
  return request.get('/letter/verification/verify', {
    params: { no: applicationNo, code }
  })
}

/**
 * 获取函件验证信息
 * @param applicationNo 函件申请编号
 */
export function getVerificationInfo(applicationNo: string): Promise<ApiResponse<LetterVerificationResult>> {
  return request.get(`/letter/verification/info/${applicationNo}`)
}

// ========== 管理后台接口 ==========

/**
 * 分页查询验证记录
 */
export function getVerificationList(params: {
  page?: number
  pageSize?: number
  status?: string
  keyword?: string
}): Promise<ApiResponse<VerificationListResponse>> {
  return request.get('/api/admin/letter-verification/list', { params })
}

/**
 * 获取验证详情
 */
export function getVerificationDetail(id: number): Promise<ApiResponse<LetterVerificationRecord>> {
  return request.get(`/api/admin/letter-verification/${id}`)
}

/**
 * 撤销验证
 */
export function revokeVerification(id: number): Promise<ApiResponse<void>> {
  return request.delete(`/api/admin/letter-verification/${id}`)
}

/**
 * 获取验证统计
 */
export function getVerificationStatistics(): Promise<ApiResponse<VerificationStatistics>> {
  return request.get('/api/admin/letter-verification/statistics')
}

File: backend/src/main/java/com/clientservice/domain/entity/LetterVerification.java
package com.clientservice.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.clientservice.common.base.BaseEntity;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * 函件验证实体 - 存储律所系统推送的电子函件验证信息
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("letter_verification")
public class LetterVerification extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 律所系统函件ID */
    private Long letterId;

    /** 来源 API Key ID */
    private Long sourceApiKeyId;

    /** 函件申请编号 */
    private String applicationNo;

    /** 验证码 */
    private String verificationCode;

    /** 函件类型编码 */
    private String letterType;

    /** 函件类型名称 */
    private String letterTypeName;

    /** 接收单位 */
    private String targetUnit;

    /** 出函律师姓名（多人逗号分隔） */
    private String lawyerNames;

    /** 律师事务所名称 */
    private String firmName;

    /** 关联项目名称（脱敏） */
    private String matterName;

    /** 审批时间 */
    private LocalDateTime approvedAt;

    /** 打印/盖章时间 */
    private LocalDateTime printedAt;

    /** 有效期截止时间 */
    private LocalDateTime validUntil;

    /** 备注 */
    private String remark;

    /** 验证次数 */
    private Integer verifyCount;

    /** 最后验证时间 */
    private LocalDateTime lastVerifyAt;

    /** 最后验证IP */
    private String lastVerifyIp;

    /** 状态 */
    private String status;

    // ========== 状态常量 ==========
    /** 状态：有效 */
    public static final String STATUS_ACTIVE = "ACTIVE";

    /** 状态：已过期 */
    public static final String STATUS_EXPIRED = "EXPIRED";

    /** 状态：已撤销 */
    public static final String STATUS_REVOKED = "REVOKED";
}
